docs/root/api-docs/xds_protocol.rst

```diff
@@ -427,20 +427,41 @@ names becomes empty, that means that the client is no longer interested in any r
+name (whether it be "*" or any other name), then this legacy semantic is no longer available; at
+that point, clearing the list of subscribed resources is interpretted as an unsubscription (see
+:ref:`Unsubscribing From Resources<xds_protocol_unsubscribing>`) rather than as a subscription
+to "*".
+
+For example, in SotW:
+- Client sends a request with :ref:`resource_names <envoy_v3_api_field_service.discovery.v3.DiscoveryRequest.resource_names>` unset. Server interprets this as a subscription to "*".
+- Client sends a request with :ref:`resource_names <envoy_v3_api_field_service.discovery.v3.DiscoveryRequest.resource_names>` set to "*" and "A". Server interprets this as continuing the existing subscription to "*" and adding a new subscription to "A".
+- Client sends a request with :ref:`resource_names <envoy_v3_api_field_service.discovery.v3.DiscoveryRequest.resource_names>` set to "A". Server interprets this as unsubscribing to "*" and continuing the existing subscription to "A".
+- Client sends a request with :ref:`resource_names <envoy_v3_api_field_service.discovery.v3.DiscoveryRequest.resource_names>` unset. Server interprets this as unsubscribing to "A" (i.e., the client has now unsubscribed to all resources). Although this request is identical to the first one, it is not interpreted as a wildcard subscription, because there has previously been a request on this stream for this resource type that set the :ref:`resource_names <envoy_v3_api_field_service.discovery.v3.DiscoveryRequest.resource_names>` field.
+
+And in incremental:
+- Client sends a request with :ref:`resource_names_subscribe <envoy_v3_api_field_service.discovery.v3.DeltaDiscoveryRequest.resource_names_subscribe>` unset. Server interprets this as a subscription to "*".
+- Client sends a request with :ref:`resource_names_subscribe <envoy_v3_api_field_service.discovery.v3.DeltaDiscoveryRequest.resource_names_subscribe>` set to "A". Server interprets this as continuing the existing subscription to "*" and adding a new subscription to "A".
+- Client sends a request with :ref:`resource_names_unsubscribe <envoy_v3_api_field_service.discovery.v3.DeltaDiscoveryRequest.resource_names_unsubscribe>` set to "*". Server interprets this as unsubscribing to "*" and continuing the existing subscription to "A".
+- Client sends a request with :ref:`resource_names_unsubscribe <envoy_v3_api_field_service.discovery.v3.DeltaDiscoveryRequest.resource_names_unsubscribe>` set to "A". Server interprets this as unsubscribing to "A" (i.e., the client has now unsubscribed to all resources). Although the set of subscribed resources is now empty, just as it was after the initial request, it is not interpreted as a wildcard subscription, because there has previously been a request on this stream for this resource type that set the :ref:`resource_names_subscribe <envoy_v3_api_field_service.discovery.v3.DeltaDiscoveryRequest.resource_names_subscribe>` field.
```

Comment:
reconnection will make things messy here if we dont explicitly say it's wildcard or not. 

Could you clarify on this more? 
Consider this case: 

1. Client subscribes: A, * 
2. Server sends: A, B, C, D
3. Client removes: * 
4. Server has update for B, C, -D, and new resource E, now does it send (update B, C, remove D) to the client or not?

Also noteworthy that reconnection for the above scenario would become tricky: 

5. reconnect happens, now client has A, B, C, D in the intial_resource_versions, Server will keep sending updates for A, B, C, D to the client.

Reply:
> reconnection will make things messy here if we dont explicitly say it's wildcard or not.
> 
> Could you clarify on this more?
> Consider this case:
> 
>     1. Client subscribes: A, *
> 
>     2. Server sends: A, B, C, D
> 
>     3. Client removes: *
> 
>     4. Server has update for B, C, -D, and new resource E, now does it send (update B, C, remove D) to the client or not?

I'm not sure. I would say that either "server sends (remove B, C and D)" or "server sends (remove *)" and client takes care of dropping wildcard resources from its cache.

> 
> 
> Also noteworthy that reconnection for the above scenario would become tricky:
> 
>     1. reconnect happens, now client has A, B, C, D in the intial_resource_versions, Server will keep sending updates for A, B, C, D to the client.

I'd say that once client unsubscribes from wildcard, it should drop the wildcard resources it has, so if reconnect happens, only A should be in initial resources.

Reply:
> Consider this case:
> 
> 1. Client subscribes: A, *
> 2. Server sends: A, B, C, D
> 3. Client removes: *
> 4. Server has update for B, C, -D, and new resource E, now does it send (update B, C, remove D) to the client or not?

As per the clarification I added in #17983, the server does not need to send anything for B, C, D, or E in this case.

> Also noteworthy that reconnection for the above scenario would become tricky:
> 
> 5. reconnect happens, now client has A, B, C, D in the intial_resource_versions, Server will keep sending updates for A, B, C, D to the client.

Once the client reconnects, it sends only A in `resource_names_subscribe`, so this is not an issue.  Even if the other resources were somehow present in `initial_resource_versions`, that field does not affect the set of resources that the client is actually subscribed to.